pkg/apiserver/watch.go

```diff
@@ -83,39 +84,51 @@ func isWebsocketRequest(req *http.Request) bool {
 
 // ServeHTTP processes watch requests.
 func (h *WatchHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	var verb string
+	var apiResource string
+	var httpCode int
+	reqStart := time.Now()
+	defer func() { monitor("watch", verb, apiResource, httpCode, reqStart) }()
+
 	if req.Method != "GET" {
 		notFound(w, req)
+		httpCode = http.StatusNotFound
 		return
 	}
 
 	requestInfo, err := h.apiRequestInfoResolver.GetAPIRequestInfo(req)
 	if err != nil {
 		notFound(w, req)
+		httpCode = http.StatusNotFound
 		return
 	}
+	verb = requestInfo.Verb
 	ctx := api.WithNamespace(api.NewContext(), requestInfo.Namespace)
 
 	storage := h.storage[requestInfo.Resource]
 	if storage == nil {
 		notFound(w, req)
+		httpCode = http.StatusNotFound
```

Comment:
this makes me slightly worries that in the future changes we'll forget to add this annotation